### DISCUSSION TITLE
Enabling bitcode

### DISCUSSION THREAD

**Discussion by ghost** [swiftyfinch/Rugby#8] [Ideas]
Hello! Thank you for your work on Rugby, it's the best tool I have tried so far for caching Pods. Would it be possible to build the Pods with bitcode, or to enable bitcode in a future release?

**Comment by swiftyfinch** [comment]
Hello, thanks for your feedback! I appreciate it 🎉 

Yeah, I already thought about it just today. But I need some time to think about the Rugby interface.
I will be back soon with more info or with the new release.

**Reply by swiftyfinch** [reply]
New release #9 was shipped! 🚢

Now if your pods using bitcode it also will be used via Rugby.
Or you can enable bitcode in this way in your Pods project targets:
```ruby
# Podfile
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      # Use bitcode for all pods except one with XCTest framework dependency
      if target.name != 'iOSSnapshotTestCase'
        config.build_settings['ENABLE_BITCODE'] = 'YES'
      end
    end
  end
end
```

@TheoGh Does this solve your problem?

**Reply by ghost** [reply]
Thank you for the quick update! Sadly there seems to be something wrong still, I get the following error when archiving for AppStore:

```
Bitcode bundle could not be generated because '/Users/[snip]/.rugby/build/Debug-iphoneos/Alamofire/Alamofire.framework/Alamofire' was built without full bitcode. All frameworks and dylibs for bitcode must be generated from Xcode Archive or Install build file '/Users/[snip]/.rugby/build/Debug-iphoneos/Alamofire/Alamofire.framework/Alamofire' for architecture arm64
```

I'm not sure at the moment if it's something in our project configuration or in Alamofire specifically. I have already added
```
config.build_settings['ENABLE_BITCODE'] = 'YES'
config.build_settings['BITCODE_GENERATION_MODE'] = 'bitcode'
```
to the Podfile. Any suggestions are appreciated.

**Reply by swiftyfinch** [reply]
Oh, I understood. It's because Rugby builds for debugging.
Are you sure you want to make archives with Rugby? Is it difficult to do it without?

I can add `-config Release` option, but I'm not sure right now that it will be enough.

**Reply by ghost** [reply]
Obviously it would be convenient to be able to make archives without rerunning `pod install`, but from my point of view this is merely a suggestion - Rugby is a tool to be used during development first and foremost, and that works now. :)

**Reply by swiftyfinch** [reply]
I investigated this case. Rugby using `xcodebuild build` command, but for the archive you need to use `xcodebuild archive`.
Archiving application is a special case and the easiest way just not to use Rugby for that.

Sorry, I can't support also `archive` option **now**. The main goal of 🏈 Rugby is to make regular daily work easier. Like debugging and developing something. But not for a release process.

**Reply by swiftyfinch** [reply]
> Obviously it would be convenient to be able to make archives without rerunning `pod install`

Yeah, that's true. But as I understand you anyway will need to run `archive` as separate command even via Rugby.
Thanks for the suggestion 👍🏻  I will put it in Rugby backlog, but as a low priority feature.

**Comment by swiftyfinch** [comment]
@TheoGh #102

**Reply by ghost** [reply]
Many thanks! I'll try it out ASAP.
